Content.Server/Jobs/AddComponentSpecial.cs

```diff
@@ -11,16 +12,27 @@ namespace Content.Server.Jobs
     public sealed class AddComponentSpecial : JobSpecial
     {
         // TODO: Type serializer that ensures the component exists.
-        [DataField("component", required:true)]
-        public string Component { get; } = string.Empty;
+        [DataField("components")]
+        [AlwaysPushInheritance]
+        public EntityPrototype.ComponentRegistry Components { get; } = new();
 
         public override void AfterEquip(IEntity mob)
         {
             // Yes, this will throw if your component is invalid.
-            var component = (Component)IoCManager.Resolve<IComponentFactory>().GetComponent(Component);
-            component.Owner = mob;
+            // now its a registry of components, still throws i bet.
+            var factory = IoCManager.Resolve<IComponentFactory>();
+            var entityManager = IoCManager.Resolve<IEntityManager>();
+            var serializationManager = IoCManager.Resolve<ISerializationManager>();
+
+            foreach (var (name, data) in Components)
+            {
+                var component = (Component) factory.GetComponent(name);
+                component.Owner = mob;
+                _ = serializationManager.Copy(data, component, null);
```

Comment:
use the return value to call addcomponent with

Reply:
return value of which function

Reply:
Copy, the one you are currently writing into _